CMakeLists.txt

```diff
@@ -124,9 +133,12 @@ set(COMMON_CXX_FLAGS
 )
 
 # Option for enabling/disabling SIMD flags is for both of debug and release
-if (NTTEC_USE_SIMD)
+if (NTTEC_USE_SIMD STREQUAL "SSE4")
   SET(COMMON_CXX_FLAGS "${COMMON_CXX_FLAGS} -msse4.1")
-  add_definitions(-DNTTEC_USE_SIMD)
+  add_definitions(-DNTTEC_USE_SIMD=1)
+elseif (NTTEC_USE_SIMD STREQUAL "AVX2")
+  SET(COMMON_CXX_FLAGS "${COMMON_CXX_FLAGS} -mavx2 -msse4.1")
```

Comment:
Is sse4.1 needed?

Activating AVX2 doesn't include SSE4.1 automatically?

Reply:
When -msse4.1 is not here, there are errors: Illegal instruction. So I keep it like that. A further investigation could be useful to figure it out.